tableau_utilities/scripts/merge_config.py

```diff
@@ -61,115 +50,36 @@ def merge_2_configs(existing_config, additional_config, debugging_logs=False):
-            if 'calculation' in column_details:
-                calculation_current = existing_config[column_name]['calculation']
-                calculation_new = column_details['calculation']
-
-                if len(calculation_new.strip()) > 0:
-                    if debugging_logs:
-                        print('CHANGING CALCULATION')
-                        print('CALCULATION CURRENT:', calculation_current)
-                        print('CALCULATION NEW:', calculation_new)
-
-                    existing_config[column_name]['calculation'] = calculation_new
-
-                    if debugging_logs:
-                        print('CALCULATION SET TO:', existing_config[column_name]['calculation'])
-
-            # Replace these attributes if there are values in the new configuration
-
-            if (isinstance(description_new, list) and len(description_new) > 0) \
-                    or (isinstance(description_new, str) and len(description_new.strip()) > 0):
-                if debugging_logs:
-                    print('CHANGING DESCRIPTION')
-                    print('DESCRIPTION CURRENT:', description_current)
-                    print('DESCRIPTION NEW:', description_new)
-
-                existing_config[column_name]['description'] = description_new
-
-                if debugging_logs:
-                    print('DESCRIPTION SET TO:', existing_config[column_name]['description'])
-
-            if folder_new is not None and len(folder_new.strip()) > 0:
-                if debugging_logs:
-                    print('CHANGING FOLDER')
-                    print('FOLDER CURRENT:', folder_current)
-                    print('FOLDER NEW:', folder_new)
-
-                existing_config[column_name]['folder'] = folder_new
-
-                if debugging_logs:
-                    print('FOLDER SET TO:', existing_config[column_name]['folder'])
-
-            if len(persona_new.strip()) > 0:
-                if debugging_logs:
-                    print('CHANGING PERSONA')
-                    print('PERSONA CURRENT:', persona_current)
-                    print('PERSONA NEW:', persona_new)
-
-                existing_config[column_name]['persona'] = persona_new
-
-                if debugging_logs:
-                    print('PERSONA SET TO:', existing_config[column_name]['persona'])
-
-            datasources_list = []
-
-            if debugging_logs:
-                print('CHANGING DATASOURCES')
-                print('DATASOURCES CURRENT:', datasources_current)
-                print('DATASOURCES NEW:', datasources_new)
-
-            datasources_names_current = [d['name'] for d in datasources_current]
+                print('EXISTING DETAIL:', existing_config[column_name])
+                print('ADDITIONAL DETAIL:', column_details)
 
+            # Replace existing config attributes
+            for k, v in column_details.items():
+                if v and k != 'datasources':
```

Comment:
This line effectively does the same things as your conditions like
```
folder_new is not None and len(folder_new.strip()) > 0
```
If `v` is None or and empty string, `if v` will effectively be false. Also, here I am intently not altering the `datasources` attribute the same as the others, since it's a list and has special logic to be updated (which I handle below)

I do however see that I am not striping the string, so it's possible `'  '` could replace things - so I will fix that. Also, I could definitely improve the debugging logs here, so I will work on that too! :)